plugins_examples/archive/archive.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+	"regexp"
+)
+
+func main() {
+	s := bufio.NewScanner(os.Stdin) 
+	for s.Scan() {
+		line := s.Text()
+		if strings.Contains(line, "PRIVMSG") == true {
+			skip := false;
+
+	    		re := regexp.MustCompile("https?\\:\\/\\/[[:^space:]]*")
+	    		match := re.FindString(line)
+
+			if match != "" && match != "https://arch.cccp.io/urls.txt" {
+	    			fr, err := os.Open("/var/www/html/arch/urls.txt")
+	    			if err != nil {
+				    fmt.Fprintln(os.Stderr, "error", err)
+	    			}
+				defer fr.Close()
+				scanner := bufio.NewScanner(fr)
+  				for scanner.Scan() {
+					if strings.Contains(scanner.Text(), match) {
+						skip = true
+						fmt.Fprintf(os.Stdout, "PRIVMSG #esil :url déjà enregistrée\r\n")	
```

Comment:
I think this is useful during debugging, but has to be deactivated otherwise
(annoying to have a comment "already there" everytime someone post an url)

OR maybe

If the URL has already been sent and more than N days, the plugin could say:  "Old! Already sent by $nickname 10 days ago."

Reply:
Removed it !